Virtual Escape Project

MAbdullah-Riaz opened the pull request:
This pull request includes updates to the active routes, including the "main-menu", "login", and "leaderboard". The project uses the React framework and this pull request includes the addition of extra features such as private routes.

The main-menu route has been updated to provide a more intuitive and user-friendly interface for navigating the game. The login route has been improved to include enhanced security measures to protect player information. The leaderboard route has been updated to accurately display the latest scores and rankings for players.

In addition to these updates, the pull request includes the implementation of private routes. This feature will allow players to access exclusive in-game content and keep their progress private from other players. The private routes will be securely protected and accessible only to authorized players.

MAbdullah-Riaz committed:
Virtual Escape project using private routes and contexts

bilalashraf1710 commented on src/components/nav-button/nav-button.styles.scss:
```diff
@@ -0,0 +1,14 @@
+button {
```
styles.scss?? should be module.scss

bilalashraf1710 commented on src/context/AuthContainer.jsx:
```diff
@@ -0,0 +1,21 @@
+import { useState, useContext } from 'react';
```
should be in a folder name AuthContext

bilalashraf1710 commented on src/routes/PrivateRoutes.jsx:
```diff
@@ -0,0 +1,11 @@
+import { Outlet, Navigate } from 'react-router-dom';
```
Can we move routes folder to container folder

bilalashraf1710 commented on src/routes/PrivateRoutes.jsx:
```diff
@@ -0,0 +1,11 @@
+
+const PrivateRoutes = () => {
+  const { isAuthenticated } = useAuthContext();
+  console.log('authenticated', isAuthenticated);
```
remove logs

bilalashraf1710 commented on src/routes/RoutesConstants.jsx:
```diff
@@ -0,0 +1,18 @@
+  return (
+    <div>
+      <Routes>
+        <Route element={<PrivateRoutes />}>
```
Create a contants file in routes folder and declare these paths as a const

bilalashraf1710 replied:
main menu path should b '/' not 'mainMenu'.
For paths we follow kebab case like main-menu

bilalashraf1710 commented on src/components/main-button/main-button.jsx:
```diff
@@ -0,0 +1,19 @@
+import './main-button.styles.scss';
```
icons, buttons, cards should be in shared folder?

bilalashraf1710 commented on src/components/main-button/main-button.jsx:
```diff
@@ -0,0 +1,19 @@
+import './main-button.styles.scss';
+
+const MainButton = ({ buttonType, name, image, className, onClickHandler }) => {
+  const onClickHandle = () => {
+    onClickHandler?.();
+  };
+  return (
+    <button
```
the name should be Button not MainButton

Shazil2154 commented on src/container/login/login.jsx:
```diff
@@ -0,0 +1,62 @@
+import styles from './login.module.scss';
+const Login = () => {
+  const { isAuthenticated, setLogin } = useAuthContext();
+  console.log('login page', isAuthenticated);
```
Never leave log statement in a pr

Shazil2154 commented on src/components/card-list/card-list.styles.scss:
```diff
@@ -0,0 +1,14 @@
+@media screen and (max-device-width: 1300px) {
+& {
+ max-width: 900px;
+//  width: 900px;
```
is this comment still needed?

Shazil2154 commented on src/components/icon/icon.styles.jsx:
```diff
@@ -0,0 +1,20 @@
+import { Tooltip } from 'react-tooltip';
```
I think we should stick to one styling library either use styles component or sass

Shazil2154 commented on src/components/return-button/return-button.jsx:
```diff
@@ -0,0 +1,13 @@
+import styles from './return-button.module.scss';
+import returnMenu from '../../assets/returnMenu.svg';
+export const ReturnButton = ({ returnButtonName, onClickhandler }) => {
+  const handleOnClick = () => {
```
why do we have to optionally chain this? In which case this will be null

Shazil2154 commented on src/container/leaderboard/leaderboard.jsx:
```diff
@@ -0,0 +1,29 @@
+        </div>
+
+        <ReturnButton
+          returnButtonName={'RETURN TO MAIN MENU'}
```
we don't have to place the string in curly braces.

Shazil2154 commented on src/container/login/login.jsx:
```diff
@@ -0,0 +1,62 @@
+  const onClickHandler = () => {
+    if (isLenMaxed) {
+      setLogin();
+      console.log('login button', isAuthenticated);
```
same here

Shazil2154 commented on src/container/main-menu/mainMenu.module.scss:
```diff
@@ -0,0 +1,23 @@
+ h1 {
+           font-size: clamp(2rem, 3.1vw, 3.75rem);  //60px
+            margin-top: 0px;
+            padding: 0 10px;
```
fix formating

Shazil2154 commented on src/shared/components/table-header/table-header.jsx:
```diff
@@ -0,0 +1,13 @@
+
+export const TableHeader = () => {
+  return (
+    <tr>
```
IMHO I think headers should be an array that is passed to this component and it maps the array and renders it accordingly. that would make this component truly reusable

Shazil2154 commented on src/shared/components/table/table.jsx:
```diff
@@ -0,0 +1,130 @@
+      rank: 18,
+      teamName: 'Noodley Doodles',
+      hints: 3,
+      incorrect: 3,
```
move dummy data to a different file

Shazil2154 reviewed (changes requested):
Added better pr descriptions

MAbdullah-Riaz committed:
Updated comments.

MAbdullah-Riaz committed:
added curser on all buttons except disable one.

MAbdullah-Riaz committed:
Added authentication in localstorage

MAbdullah-Riaz committed:
Removed all warning and removed access of logIn page if user is logged In

bilalashraf1710 commented on src/container/routes/RoutesConstants.jsx:
```diff
@@ -0,0 +1,23 @@
+import { Route, Routes } from 'react-router-dom';
```
File name should be Routes

bilalashraf1710 commented on src/container/leaderboard/leaderboard.jsx:
```diff
@@ -0,0 +1,32 @@
+import styles from './Leaderboard.module.scss';
```
file name should b Pascal

bilalashraf1710 commented on src/container/navigation/navigation.jsx:
```diff
@@ -0,0 +1,39 @@
+import experiosLogo from 'assets/logo/logo1.png';
```
inconsistent file naming
Pascal for scss file
camel for jsx file

MAbdullah-Riaz replied:
I think you meant pascal for jsx and camel for scss. Right?

bilalashraf1710 commented on src/container/routes/constants.js:
```diff
@@ -0,0 +1,3 @@
+export const MAIN_ROUTE = '/';
```
should b a pascal file name.
Push your code after finalizing stuff

MAbdullah-Riaz replied:
I did convert it into Pascal case all the file but its not updating on github just test it on test commit , github ignores the capitalization of filename (just asked from Shazil).

MAbdullah-Riaz replied:
constants.js is not the JSX, I asked from Ahmed about the naming convention of js files,as he was suggested to name it like that (constants.js) thats why I also name js files in the data file camels cases, from now on I'll go with Pascal

bilalashraf1710 commented on src/shared/components/button/Button.jsx:
```diff
@@ -0,0 +1,25 @@
+      onClick={onClickHandle}
+      className={`${styles['button-container']} ${styles[buttonTypeClass]} ${styles[className]} `}
+    >
+      {leftImage} {name} {rightImage}
```
this is wrong Abdullah, we need to use ternary check for icons

MAbdullah-Riaz committed:
test-filename

bilalashraf1710 commented on src/container/login/login.jsx:
```diff
@@ -1,15 +1,15 @@
 import Input from 'components/input/Input';
 import MainButton from 'shared/components/button/Button';
-import experiousLogo from 'assets/logo/logo1.png';
+import ExperiousLogo from 'assets/logo/logo1.png';
```
Please fix the order or your imports, ask Shazil or Ahmad about what the order of imports should be

MAbdullah-Riaz committed:
Changing names pt.1

MAbdullah-Riaz committed:
Changing names pt.2

MAbdullah-Riaz committed:
Changing names pt. 3

MAbdullah-Riaz committed:
Changed Names

MAbdullah-Riaz committed:
Updated Folder Structure

bilalashraf1710 commented on src/container/login/Login.jsx:
```diff
@@ -0,0 +1,65 @@
+  const { setLogin } = useAuthContext();
+
+  const navigate = useNavigate();
+  const [isLenMaxed, setIsLenMaxed] = useState(false);
```
whats the purpose of this?
you just need to disable button and the class of disable button should work through suedo selector

bilalashraf1710 commented on src/container/login/Login.jsx:
```diff
@@ -0,0 +1,65 @@
+  };
+  return (
+    <div className={styles['main-login-container']}>
+      <div className={styles['logo-container']}>
```
this is part of Navigation? we should have props for navigation where we can mention what to render.
If we want to render only logo then shouldLogoRender = true

bilalashraf1710 commented on src/container/login/Login.jsx:
```diff
@@ -0,0 +1,65 @@
+              buttonType={'log-in'}
+              name={'login'}
+              rightImage={<ThumbIcon />}
+              className={`${isLenMaxed ? 'len-maxed' : ''}`}
```
should be is disable true

bilalashraf1710 commented on src/container/routes/Routes.jsx:
```diff
@@ -0,0 +1,23 @@
+        <Route path={LEADERBOARD_ROUTE} element={<Leaderboard />} />
+        <Route index path={MAIN_ROUTE} element={<MainMenu />} />
+      </Route>
+      <Route path={LOGIN_ROUTE} element={<LogIn />} />
```
What if users are authenticated and they want to hit the login page by inputting the path in the address bar? they should navigate to index page.

bilalashraf1710 replied:
What is someone enter /afbdafdkjafda? that user should navigate to index page

bilalashraf1710 commented on src/context/auth-context/AuthContainer.jsx:
```diff
@@ -0,0 +1,26 @@
+
+const AuthContextContainer = ({ children }) => {
+  const [isAuthenticated, setIsAuthenticated] = useState(
+    JSON.parse(localStorage.getItem('isAuthenticated')) ?? false
```
We should do this in UseEffect.

bilalashraf1710 commented on src/context/auth-context/AuthContainer.jsx:
```diff
@@ -0,0 +1,26 @@
+  const [isAuthenticated, setIsAuthenticated] = useState(
+    JSON.parse(localStorage.getItem('isAuthenticated')) ?? false
+  );
+  const setLogin = () => {
```
We should pass token as a parameter to this function from login page.

MAbdullah-Riaz replied:
we currently don't have a BE to generate an auth token. So I guess there is no need to store the token if that's what you mean. Otherwise, can you kindly explain?

MAbdullah-Riaz committed:
fix comments

added navigation logoOnlyRender as a prop and imported in the login page.
added disable prop in login page
added classes as prop in Button component
handle the input path of login by url
handle random input typed in URL.

MAbdullah-Riaz committed:
fix comments

Updated dependency array in ueEffect in login page
added token from the login page (temporary added in argument ) and stored it on local storage.
fix folder of components of leaderboard and main-menu

MAbdullah-Riaz committed:
Added axios and auth context

MAbdullah-Riaz committed:
added token in privateRoutes
